### PR TITLE
Import BBS archive with migrate-repo command

### PR DESCRIPTION
**Pull request by synthead**
Closes https://github.com/github/gh-gei/issues/528!

This PR updates the placeholder `migrate-repo` command for `bbs2gh` with real logic to ingest an archive URL and kick off a migration with Octoshift.

A few notes about this PR:

- There are some parameters that pass strings like `"not-used"` to satisfy required parameters in GraphQL and Octoshift.  When https://github.com/github/octoshift/issues/5350 is shipped, these "filler" strings can be removed.
- `CreateBbsMigrationSource` was added to `GithubApi`.  This is consumed by the bbs2gh `MigrateRepoCommand` to kick off migrations over GraphQL.
- `bbs2gh.csproj` was added as a reference to `OctoshiftCLI.Tests.csproj`.  This is necessary to test classes in the `OctoshiftCLI.BbsToGithub.Commands` namespace.
- I opted to use a `MigrateRepoCommandArgs` class in bbs2gh's `MigrateRepoCommand` because I felt that it was cleaner and easier to read.
- Some unused ADO logic was pulled from bbs2gh's `EnvironmentVariableProvider`.
- As for test coverage, I opted to find parity in the coverage for [the GEI tests](https://github.com/github/gh-gei/blob/31fc7fe08b964778eec4e3b35117bda405395bef/src/OctoshiftCLI.Tests/gei/Commands/MigrateRepoCommandTests.cs) that is relevant to bbs2gh.  I'm happy to add more coverage if desired 👍 

## Tasks

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created) (added bullet point to https://github.com/github/gh-gei/issues/586)

### CODE REVIEW

File: src/bbs2gh/Commands/MigrateRepoCommand.cs [hunk 1]
```diff
@@ -14,27 +14,21 @@ public class MigrateRepoCommand : Command
...
-            };
-            var adoTeamProject = new Option<string>("--ado-team-project")
-            {
-                IsRequired = true
-            };
-            var adoRepo = new Option<string>("--ado-repo")
-            {
-                IsRequired = true
+                IsRequired = true,
+                Description = "URL used to downlodad Bitbucket Server migration archive."
```
Comment (dylan-smith) [inline src/bbs2gh/Commands/MigrateRepoCommand.cs:31]: typo in download

File: src/bbs2gh/Commands/MigrateRepoCommand.cs [hunk 2]
```diff
@@ -44,98 +38,92 @@ public class MigrateRepoCommand : Command
...
-            };
-            var wait = new Option("--wait")
+            var githubPat = new Option<string>("--github-pat")
             {
-                IsRequired = false,
-                Description = "Synchronously waits for the repo migration to finish."
+                IsRequired = false
             };
-            var adoPat = new Option<string>("--ado-pat")
+            var githubApiUrl = new Option<string>("--github-api-url")
```
Comment (dylan-smith) [inline src/bbs2gh/Commands/MigrateRepoCommand.cs:45]: I don't think we should include this arg. This is primarily to support GHAE, which I don't want to do right now (hopefully never).

File: src/bbs2gh/Commands/MigrateRepoCommand.cs [hunk 3]
```diff
@@ -44,98 +38,92 @@ public class MigrateRepoCommand : Command
...
+            var githubApi = _githubApiFactory.Create(apiUrl: args.GithubApiUrl, targetPersonalAccessToken: args.GithubPat);
+            var githubOrgId = await githubApi.GetOrganizationId(args.GithubOrg);
+            var migrationSourceId = await githubApi.CreateBbsMigrationSource(githubOrgId);
 
             string migrationId;
 
             try
             {
-                migrationId = await githubApi.StartMigration(migrationSourceId, adoRepoUrl, githubOrgId, githubRepo, adoPat, githubPat);
+                migrationId = await githubApi.StartMigration(
```
Comment (dylan-smith) [inline src/bbs2gh/Commands/MigrateRepoCommand.cs:104]: could we move the knowledge about not-used stuff down into githubApi class. For example, maybe have a 2nd overload of startMigration that doesn't have args for the 3 not-used things, then in that function put the dummy values in the api payload.

File: src/bbs2gh/Program.cs [hunk 4]
```diff
@@ -47,7 +47,7 @@ public static async Task Main(string[] args)
             }
             catch (Exception ex)
             {
-                Logger.LogWarning("Could not retrieve latest gei CLI version from github.com, please ensure you are using the latest version by running: gh extension upgrade ado2gh");
+                Logger.LogWarning("Could not retrieve latest gei CLI version from github.com, please ensure you are using the latest version by running: gh extension upgrade bbs2gh");
```
Comment (dylan-smith) [inline src/bbs2gh/Program.cs:50]:
```suggestion
                Logger.LogWarning("Could not retrieve latest bbs2gh extension version from github.com, please ensure you are using the latest version by running: gh extension upgrade bbs2gh");
```

File: src/bbs2gh/Program.cs [hunk 5]
```diff
@@ -66,18 +66,18 @@ private static async Task LatestVersionCheck(ServiceProvider sp)
 
             if (await versionChecker.IsLatest())
             {
-                Logger.LogInformation($"You are running the latest version of the ado2gh CLI [v{await versionChecker.GetLatestVersion()}]");
+                Logger.LogInformation($"You are running the latest version of the bbs2gh CLI [v{await versionChecker.GetLatestVersion()}]");
```
Comment (dylan-smith) [inline src/bbs2gh/Program.cs:69]:
```suggestion
                Logger.LogInformation($"You are running the latest version of the bbs2gh extension [v{await versionChecker.GetLatestVersion()}]");
```

File: src/bbs2gh/Commands/MigrateRepoCommand.cs (hunk 1 above)
Comment (synthead) [reply src/bbs2gh/Commands/MigrateRepoCommand.cs]: Good catch! Fixed in 52f63c12bb33b397cdc567dd3854a7fb2e3ad0bf!

File: src/bbs2gh/Commands/MigrateRepoCommand.cs (hunk 2 above)
Comment (synthead) [reply src/bbs2gh/Commands/MigrateRepoCommand.cs:45]: After talking about this in a 🍐ing session, we decided to take this option out and include it as an environment variable in the future.  This makes it easier to test in development, but for customers, it doesn't really serve a very useful feature, and adds code complexity 👍

Comment (synthead) [reply src/bbs2gh/Commands/MigrateRepoCommand.cs]: Removed in 273316e0a520cabc669fe582d56ea306a27d3cf4!

File: src/bbs2gh/Program.cs (hunk 4 above)
Comment (synthead) [reply src/bbs2gh/Program.cs]: Fixed in a2c57ef014d977a45328d459db90aabee09bf801!

File: src/bbs2gh/Program.cs (hunk 5 above)
Comment (synthead) [reply src/bbs2gh/Program.cs]: Fixed in b6f8dce93e94a2d605390f0d2975e50b57c52716!  Found another one, too, so I also included it in this commit 👍

File: src/bbs2gh/Program.cs [hunk 6]
```diff
@@ -66,18 +66,18 @@ private static async Task LatestVersionCheck(ServiceProvider sp)
...
             }
             else
             {
-                Logger.LogWarning($"You are running an older version of the ado2gh CLI [v{versionChecker.GetCurrentVersion()}]. The latest version is v{await versionChecker.GetLatestVersion()}.");
-                Logger.LogWarning($"Please update by running: gh extension upgrade ado2gh");
+                Logger.LogWarning($"You are running an older version of the bbs2gh CLI [v{versionChecker.GetCurrentVersion()}]. The latest version is v{await versionChecker.GetLatestVersion()}.");
```
[the remark this answers is not in the document]
Comment (synthead) [reply src/bbs2gh/Program.cs]: Fixed in b6f8dce93e94a2d605390f0d2975e50b57c52716 👍

**Comment by dylan-smith** [comment]
https://github.com/github/gh-gei/issues/586 is the tracking issue for docs, that will come later when we're getting ready for private beta.

File: src/bbs2gh/Commands/MigrateRepoCommand.cs (hunk 3 above)
Comment (synthead) [reply src/bbs2gh/Commands/MigrateRepoCommand.cs]: Done!

- Added `StartBbsMigration` to `GithubApi` with 88113db3bd988c43deced584512c2c0548ff65b1
- Made `MigrateRepoCommand` call `StartBbsMigration` with cce2213d0a3df9a176c660b5db27b8327d5e057c

**Review by dylan-smith — APPROVED** [review]

**Pull request merged**